src/kagglehub/exceptions.py

```diff
@@ -53,3 +53,11 @@ def kaggle_api_raise_for_status(response: requests.Response):
 
         # Default handling
         raise KaggleApiHTTPError(message, response=response) from e
+
+
+def process_post_response(response: Dict[str, Any]):
+    """
+    Postprocesses the API response to check for errors.
+    """
+    if "error" in response and response["error"] != "":
```

Comment:
Looks like when the response fails with a 500 (internal server error), here is what response looks like:

```
{'code': 500, 'message': 'InternalServerError'}
```

Make sure to check whether the code isn't a 2XX.  
And raise a BackendError if that's the case with the error code and the `message`.